docs/configuration.asciidoc

```diff
@@ -108,6 +108,26 @@ NOTE: The service name must conform to this regular expression: ^[a-zA-Z0-9 _-]+
 | Name of the entry assembly       | String 
 |============
 
+[float]
+[[config-service-version]]
+==== `ServiceVersion`
+
+This is used to know exact version of your service.
+
+NOTE: The service version must conform to this regular expression: ^[0-9]+\.[0-9]+(?:\.[0-9]+)?(?:\.[0-9]+)?$. In less regexy terms: Your service version must contain only numbers (from 2 to 4), which are separated by `.` (dot).
+
+[options="header"]
+|============
+| Environment variable name      | IConfiguration key 
+| `ELASTIC_APM_SERVICE_VERSION` | `ElasticApm:ServiceVersion`
+|============
+
+[options="header"]
+|============
+| Default                          | Type
+| Version of the entry assembly    | Version 
```

Comment:
All [other Elastic APM agents](https://docs.google.com/spreadsheets/d/1JJjZotapacA3FkHc2sv_0wiChILi3uKnkwLTjtBmxwU/) don't have default value for this configuration option and it's not mandatory [APM Server API specification](https://github.com/elastic/apm-server/blob/6.5/docs/spec/service.json#L79) so I recommend to align with other agents' approach.

Reply:
I believe we need to discuss it in details. From user perspective I want to use service version to determine exact version of my application, which has some issues. And with this we don't any problems in case of Elastic.Apm. However, we need to look into set up process of service version. In my own experience, we always set proper version for assemblies during CI process (lately, [`GitVersion`](https://github.com/GitTools/GitVersion) heps us with this). In case of we don't have default value, a different approach should be defined to set service version into configuration file during build or environment variable during deploy process. But this reason, I think it's nice to have default value for service version which is fetched from executing assembly.
@SergeyKleyman, what do you think? From one side, it's different behaviour from various agents, fro other one - is good for end users.

Reply:
Yeah, this is tricky - I'd have implemented it the same way @vhatsura as you did. But usually we default to what other agents do. I think for .NET devs it'd be better to have the version of the entry assembly as default though. I asked other agent devs what they think - will report back.

One technical note: if the process is hosted by a native application (e.g. IIS) then `Assembly.GetEntryAssembly()` returns null, so this default won't work for apps directly hosted in IIS - I guess typically ASP.NET Classic, or ASP.NET Core in process - but I think that's fine.

Reply:
@gregkalapos, in case of Assembly.GetEntryAssembly() returns null, fallback to current stack trace will be involved (implemented as for service name).

Reply:
Yes, I see it, but that may have some drawbacks, see the end of this comment: https://github.com/elastic/apm-agent-dotnet/issues/50#issuecomment-455134362 

In case we do implement it with trying to have a default maybe in case of the service version we should only use it when `Assembly.GetEntryAssembly()` is not null. The service name is mandatory, so we needed to set it to something, but we know it's not always correct, in that case it can be corrected by setting it manually. Probably for an optional field skipping it would be better if we know that we just "guess" the version.

Reply:
Yeap, make sense. Will stop on `Assembly.GetEntryAssembly`

Reply:
`service.name` is [mandatory for 6.5](https://github.com/elastic/apm-server/blob/6.5/docs/spec/service.json#L55) but [optional for 7.0](https://github.com/elastic/apm-server/blob/7.0/docs/spec/service.json#L52) so if we want to make it optional we will have to increase [APM Server version in compatibility matrix](https://www.elastic.co/guide/en/apm/get-started/7.3/agent-server-compatibility.html#dotnet-compatibility).
OTOH `service.version` is [optional at least from 6.5](https://github.com/elastic/apm-server/blob/6.5/docs/spec/service.json#L81)

Reply:
I asked devs from other agents and we tend to agree that it's fine to try to report a default version this way - most of them planning to have something similar. After dc3f68d we are ok here.